pkg/api/types.go

```diff
@@ -2070,10 +2070,83 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// ComponentType constants define the known types of components.
+// Each type may have multiple instances.
+// Each instance's name is prefixed by the type, with a random suffix. See SimpleNameGenerator.
```

Comment:
Does this mean that this API object has different semantics than EVERY OTHER object?  Specifically, if I specify `metadata.name` on a Component, will you ignore or override that?  I would call that wrong.